src/dos/cdrom_ioctl_linux.cpp

```diff
@@ -113,4 +324,175 @@ bool CDROM_Interface_Ioctl::ReadSectorsHost([[maybe_unused]] void *buffer,
 	return false; /*TODO*/
 }
 
+void CDROM_Interface_Ioctl::InitAudio(const int device_number)
+{
+	if (mixer_channel) {
+		return;
+	}
+
+	std::string name = std::string(MixerChannelNamePrefix) +
+	                   std::to_string(device_number);
+
+	// Input buffer is used as a C-style buffer passed to an ioctl.
+	// Its size must always be 1 second of data.
+	// Stored in int16_t samples.
+	input_buffer.resize(REDBOOK_PCM_FRAMES_PER_SECOND * REDBOOK_CHANNELS);
+
+	// Output buffer uses C++ vector operations.
+	// Its max capacity will be 1 second so go ahead and reserve that.
+	// Stored in AudioFrame structs.
+	output_buffer.reserve(REDBOOK_PCM_FRAMES_PER_SECOND);
```

Comment:
Yep, as explained, better not to do this. Only convert as many `int16` samples from the `input_buffer` that the audio callback requested to even out performance spikes. And even then let the `output_buffer` resize itself lazily—the number of requested samples will hover around the same value.

Reply:
Reserve has been removed.  Output buffer will be much smaller now (45 samples or so instead of a full second).